packages/bot-skeleton/src/services/api/ticks_service.js

```diff
@@ -188,31 +190,27 @@ export default class TicksService {
     }
 
     observe() {
-        this.api.events.on('tick', r => {
-            const {
-                tick,
-                tick: { symbol, id },
-            } = r;
-
-            if (this.ticks.has(symbol)) {
-                this.subscriptions = this.subscriptions.setIn(['tick', symbol], id);
-                this.updateTicksAndCallListeners(symbol, updateTicks(this.ticks.get(symbol), parseTick(tick)));
+        this.api.onMessage().subscribe(({ data }) => {
+            if (data.msg_type === 'tick') {
+                const { tick } = data;
+                const { symbol, id } = data.tick;
+                if (this.ticks.has(symbol)) {
+                    this.subscriptions = this.subscriptions.setIn(['tick', symbol], id);
+                    this.updateTicksAndCallListeners(symbol, updateTicks(this.ticks.get(symbol), parseTick(tick)));
+                }
             }
-        });
 
-        this.api.events.on('ohlc', r => {
-            const {
-                ohlc,
-                ohlc: { symbol, granularity, id },
-            } = r;
-
-            if (this.candles.hasIn([symbol, Number(granularity)])) {
-                this.subscriptions = this.subscriptions.setIn(['ohlc', symbol, Number(granularity)], id);
-                const address = [symbol, Number(granularity)];
-                this.updateCandlesAndCallListeners(
-                    address,
-                    updateCandles(this.candles.getIn(address), parseOhlc(ohlc))
-                );
+            if (data.msg_type === 'ohlc') {
+                const { ohlc } = data;
+                const { symbol, granularity, id } = data.ohlc;
```

Comment:
Same here, you can get rid of line 204 if you access `symbol`, `granularity`, and `id` through `data.ohlc`.

Reply:
ohlc used directly in this function 
I update const { symbol, granularity, id } = data.ohlc; to  const { symbol, granularity, id } =ohlc;